src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,503 @@ title: Builder.io & Astro
+
+:::tip[Troubleshooting]
+
+Things can sometimes go wrong when setting up the preview. If something's not right, you can try one of these things:
+
+* Make sure the site is live - for example, your dev server is running.
+* Make sure that the URLs match exactly - the one in your Astro project and the one set in the Builder app.
+* Make sure it's the full URL including the protocol, for example `https://`.
+* If you're working in a virtual environment like [StackBlitz](https://stackblitz.com/) or [Gitpod](https://www.gitpod.io/), you might have to copy and paste the URL again when you restart your workspace, since this usually generates a new URL for your project.
+
+For more ideas, read [Builder's troubleshooting guide](https://www.builder.io/c/docs/guides/preview-url-working).
+:::
+
+### Creating a blog post
+
+1. Before you can create new content in Builder, make sure [the preview URL for your model is working](#setting-up-the-preview).
+2. Open the visual editor for a content entry of our `blogpost` model. If you've followed [the previous section](#testing-the-preview-url-setup), you can use that entry. Otherwise, create a new one: Go to the **Content** tab in your Builder space and click on **New**.
+3. In the panel on the right under the **options** tab, you should see the data fields `title` and `slug`. These are the fields you've created when you [set up your model](#creating-a-model-for-a-blog-post). Fill them in with the following values:
+    - **title:** 'First post, woohoo!'
+    - **slug:** 'first-post-woohoo'
+4. Design your post. Click on the **Add Block** button and add something - maybe a text field with some text to start with.
+5. In the text field above the editor, give your entry a name. This is how it will be listed in the Builder app.
+6. When you're ready click the **Publish** button in the upper right corner.
+
+Your content is now live in the Builder CMS but not part of your Astro project yet. You'll be fetching and displaying it in the following steps.
+
+### Displaying a list of blog posts
+
+Create the page `src/pages/index.astro` that will display a list of post titles, each linking to its own page:
+
+<FileTree title="Project Structure">
+- src/
+  - pages/
+    - **index.astro**
+  - env.d.ts
+- .env
+- astro.config.mjs
+- package.json
+</FileTree>
+
+Then add the following content:
+
+```astro title="src/pages/index.astro" /title|slug/
+---
+const builderAPIpublicKey = import.meta.env.BUILDER_API_PUBLIC_KEY;
+const builderModel = import.meta.env.BUILDER_BLOGPOST_MODEL;
+
+const { results: posts } = await fetch(
+  `https://cdn.builder.io/api/v3/content/${builderModel}?${new URLSearchParams({
+    apiKey: builderAPIpublicKey,
+    fields: ["data.slug", "data.title"].join(","),
+    cachebust: "true",
+  }).toString()}`
+)
+  .then((res) => res.json())
+  .catch();
+---
+```
```

Comment:
```suggestion
```
deleting this example, so it's not done in two steps.

The instructions above described displaying a list, but the code was only for the fetch. I think these are both straightforward enough we can combine into one code example!